src/EditorFeatures/CSharpTest/RemoveUnnecessaryParentheses/RemoveUnnecessaryParenthesesTests.cs

```diff
@@ -0,0 +1,1655 @@
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsRemoveUnnecessaryParentheses)]
+        public async Task TestConditionalExpression_TestNotAvailableForComplexChildren1()
+        {
+            await TestMissingAsync(
+@"class C
+{
+    void M()
+    {
+        var q = $$(a * b) ? (1 + 2) : (3 + 4);
+    }
+}", parameters: new TestParameters(options: RemoveAllUnnecessaryParentheses));
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsRemoveUnnecessaryParentheses)]
+        public async Task TestConditionalExpression_TestNotAvailableForComplexChildren2()
+        {
+            await TestMissingAsync(
+@"class C
+{
+    void M()
+    {
+        var q = (a * b) ? $$(1 + 2) : (3 + 4);
+    }
+}", parameters: new TestParameters(options: RemoveAllUnnecessaryParentheses));
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsRemoveUnnecessaryParentheses)]
+        public async Task TestConditionalExpression_TestNotAvailableForComplexChildren3()
+        {
+            await TestMissingAsync(
+@"class C
+{
+    void M()
+    {
+        var q = (a * b) ? (1 + 2) : $$(3 + 4);
+    }
+}", parameters: new TestParameters(options: RemoveAllUnnecessaryParentheses));
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsRemoveUnnecessaryParentheses)]
+        public async Task TestConditionalExpression_TestAvailableForPrimaryChildren1()
+        {
+            await TestAsync(
+@"class C
+{
+    void M()
+    {
+        var q = $$(a.X()) ? (1 + 2) : (3 + 4);
+    }
+}",
+@"class C
+{
+    void M()
+    {
+        var q = a.X() ? (1 + 2) : (3 + 4);
+    }
+}", offeredWhenRequireForClarityIsEnabled: true);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsRemoveUnnecessaryParentheses)]
+        public async Task TestConditionalExpression_TestAvailableForPrimaryChildren2()
+        {
+            await TestAsync(
+@"class C
+{
+    void M()
+    {
+        var q = (a.X()) ? $$(x.Length) : (3 + 4);
+    }
+}",
+@"class C
+{
+    void M()
+    {
+        var q = (a.X()) ? x.Length : (3 + 4);
+    }
+}", offeredWhenRequireForClarityIsEnabled: true);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsRemoveUnnecessaryParentheses)]
+        public async Task TestConditionalExpression_TestAvailableForPrimaryChildren3()
+        {
+            await TestAsync(
+@"class C
+{
+    void M()
+    {
+        var q = (a.X()) ? (1 + 2) : $$(a[0]);
+    }
+}",
+@"class C
+{
+    void M()
+    {
+        var q = (a.X()) ? (1 + 2) : a[0];
+    }
+}", offeredWhenRequireForClarityIsEnabled: true);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsRemoveUnnecessaryParentheses)]
+        public async Task TestIsPattern_TestAvailableWithAlwaysRemove_And_NotAvailableWhenRequiredForClarity_1()
+        {
+            await TestAsync(
+@"class C
+{
+    void M()
+    {
+        if ( $$(a[0]) is string s) { }
```

Comment:
❕ Also need tests for the following (didn't see them in a quick search):

```
if ((expression is bool b) && b) { }

if (!(expression is bool b)) { }
```

Reply:
tests added.